packages/scripts/src/addOrbitChain/transforms.ts

```diff
@@ -265,49 +272,66 @@ export const resizeImage = async (
+): Promise<{ buffer: Buffer; fileExtension: string }> => {
+  let imageBuffer: Buffer;
+  let fileExtension: string;
+
+  // Check if the URL is an IPFS URL or starts with http/https
+  if (urlOrPath.startsWith("ipfs://") || urlOrPath.startsWith("http")) {
+    // Handle remote URLs (including IPFS)
+    if (urlOrPath.startsWith("ipfs://")) {
+      urlOrPath = `https://ipfs.io/ipfs/${urlOrPath.slice(7)}`;
+    }
+
+    console.log("Fetching image from:", urlOrPath);
+
+    const response = await axios.get(urlOrPath, {
+      responseType: "arraybuffer",
+      timeout: 10000, // 10 seconds timeout
+    });
+
+    if (response.status !== 200) {
+      throw new Error(`Failed to fetch image. Status: ${response.status}`);
+    }
+
+    imageBuffer = Buffer.from(response.data);
+
+    // Try to determine file extension from response headers
+    fileExtension = lookup(response.headers["content-type"] as string) || "";
+    if (!fileExtension) {
+      // If not found in headers, try to determine from the image data
+      const detectedType = await fileTypeFromBuffer(imageBuffer);
+      fileExtension = detectedType ? `.${detectedType.ext}` : "";
+    }
+  } else {
+    // Handle local paths
     const localPath = `../../arb-token-bridge-ui/public/${urlOrPath}`;
     if (!fs.existsSync(localPath)) {
       throw new Error(
         `Provided local path '${localPath}' did not match any existing images.`
       );
     }
-    return urlOrPath;
+    imageBuffer = fs.readFileSync(localPath);
+    fileExtension = path.extname(localPath);
   }
 
-  const fileExtension = urlOrPath.split(".").pop()?.split("?")[0] || "";
-  if (!SUPPORTED_IMAGE_EXTENSIONS.includes(fileExtension)) {
-    throw new Error(
-      `Invalid image extension '${fileExtension}'. Expected: ${SUPPORTED_IMAGE_EXTENSIONS.join(
-        ", "
-      )}.`
-    );
+  // If still not found, default to .png
+  if (!fileExtension) {
+    console.warn("Could not determine file type, defaulting to .png");
+    fileExtension = ".png";
   }
 
-  const imageSavePath = `images/${fileName}.${fileExtension}`;
-  const fullSavePath = `../../arb-token-bridge-ui/public/${imageSavePath}`;
-
-  // Create directories if they don't exist
-  const dirs = fullSavePath.split("/").slice(0, -1).join("/");
-  if (!fs.existsSync(dirs)) {
-    fs.mkdirSync(dirs, { recursive: true });
-  }
-
-  if (fs.existsSync(fullSavePath)) {
-    warning(
-      `${fileName} already exists at '${imageSavePath}'. Using the existing image.`
+  if (!SUPPORTED_IMAGE_EXTENSIONS.includes(fileExtension.replace(".", ""))) {
+    console.warn(
+      `Unsupported image extension '${fileExtension}'. Converting to PNG.`
     );
-    return `/${imageSavePath}`;
-  }
 
-  const response = await axios.get(urlOrPath, { responseType: "arraybuffer" });
-  let imageBuffer = Buffer.from(response.data);
+    // Convert the image to PNG using sharp
+    imageBuffer = await sharp(imageBuffer).png().toBuffer();
+    fileExtension = ".png";
```

Comment:
we should also try to compress the images

i think we should set a resize target like maybe 300x300?

and ideally output webp since it's usually more compact and supports transparency